server.lua

```diff
@@ -412,25 +413,32 @@ function GetDiscordRoles(user, guild --[[optional]])
 	end
 
 	if discordId then
-		if Config.CacheDiscordRoles and recent_role_cache[discordId] and recent_role_cache[discordId][guildId] then
-			return recent_role_cache[discordId][guildId]
+		-- Get roles for specified guild (or main guild if not specified)
+		roles = GetUserRolesInGuild(discordId, guildId)
+		-- If specified guild or disabled multiguild option, just return this one
+		if guild or Config.Multiguild == false then
+			return roles
 		end
-		local endpoint = ("guilds/%s/members/%s"):format(guildId, discordId)
-		local member = DiscordRequest("GET", endpoint, {})
-		if member.code == 200 then
-			local data = json.decode(member.data)
-			local roles = data.roles
-			local found = true
-			if Config.CacheDiscordRoles then
-        		recent_role_cache[discordId] = recent_role_cache[discordId] or {}
-				recent_role_cache[discordId][guildId] = roles
-				Citizen.SetTimeout(((Config.CacheDiscordRolesTime or 60)*1000), function() recent_role_cache[discordId][guildId] = nil end)
+
+		-- MULTIGUILD SECTION
+		-- Keep main guild roles so we can add the rest on top of this list
+		roles = (type(roles) == "table") and roles or {}
+		-- Loop through guilds in config and get the roles from each one.
+		for _,id in pairs(Config.Guilds) do
+			-- If it's the main guild, we already fetched these roles. NEXT!
+			if id == guildId then goto skip end

```

Comment:
I have a feeling this code may not be working which is why the roles are duplicating... 🤔

Reply:
If that duplication check wasn't working, I think it would be dup'ing the roles for the main server rather than non-main servers.

Currently the symptoms indicate an issue with collecting and caching the multiguild roles non-inclusive of the primary guild.

Jesse S. wrote:
> Just the ones from the secondary server. [are duplicating]
> Disabling the Caching in the config stops that issue.